# Perbandingan Akurasi Klasifikasi

Membandingkan data ecoli akurasi dari modelling

1. Naive Bayes:
    * sebelum oversampling
    * sesudah oversampling SMOTE    
    * sesudah oversampling ADASYN

1. Random Forest:
    * sebelum oversampling
    * sesudah oversampling SMOTE    
    * sesudah oversampling ADASYN

1. Ensemble Bagging (Naive Bayes):
    * sebelum oversampling
    * sesudah oversampling SMOTE    
    * sesudah oversampling ADASYN

1. Naive Bayes (Decision Tree):
    * sebelum oversampling
    * sesudah oversampling SMOTE    
    * sesudah oversampling ADASYN

In [2]:
import pandas as pd

data = {
    "Naive Bayes": [0.955, 0.969, 0.961],
    "Random Forest": [0.897, 0.825, 0.835],
    "Bagging (Naive Bayes)": [0.911, 0.829, 0.84],
    "Bagging (Decision Tree)": [0.970, 0.965, 0.935],
}

index = [
    "Sebelum Oversampling",
    "Sesudah Oversampling SMOTE",
    "Sesudah Oversampling ADASYN",
]

df = pd.DataFrame(data, index=index)
df


Naive Bayes  Random Forest  \
Sebelum Oversampling               0.955          0.897   
Sesudah Oversampling SMOTE         0.969          0.825   
Sesudah Oversampling ADASYN        0.961          0.835   

                             Bagging (Naive Bayes)  Bagging (Decision Tree)  
Sebelum Oversampling                         0.911                    0.970  
Sesudah Oversampling SMOTE                   0.829                    0.965  
Sesudah Oversampling ADASYN                  0.840                    0.935

## **Kesimpulan**
Berdasarkan hasil pengujian, dapat disimpulkan bahwa:

Sebelum dilakukan oversampling, metode Bagging Decision Tree memberikan performa tertinggi dengan akurasi 0.970, diikuti oleh Naive Bayes (0.955) dan Bagging Naive Bayes (0.911). Random Forest berada di posisi terendah (0.897).

Setelah oversampling menggunakan SMOTE, akurasi Naive Bayes meningkat menjadi 0.969 (lebih tinggi dibanding sebelum oversampling). Sementara itu, Bagging Decision Tree tetap stabil tinggi (0.965), meskipun sedikit turun dibanding kondisi awal. Random Forest justru mengalami penurunan performa cukup signifikan (0.825).

Dengan oversampling ADASYN, tren yang mirip terlihat: Naive Bayes tetap unggul (0.961), sedangkan Bagging Decision Tree tetap konsisten baik (0.935). Random Forest kembali lebih rendah (0.835) dan Bagging Naive Bayes sedikit naik (0.840).